compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeWindowDelegate.desktop.kt

```diff
@@ -72,18 +72,37 @@ internal class ComposeWindowDelegate(
         }
 
         override fun add(component: Component): Component {
-            val clipComponent = ClipComponent(component)
-            clipMap[component] = clipComponent
-            bridge.clipComponents.add(clipComponent)
-            return add(component, Integer.valueOf(0))
+            addToLayer(component, componentLayer)
+            return component
         }
 
-        override fun remove(component: Component) {
-            bridge.clipComponents.remove(clipMap[component]!!)
-            clipMap.remove(component)
-            super.remove(component)
+        private fun addBridge(bridge: ComposeBridge) {
+            addToLayer(bridge.invisibleComponent, bridgeLayer)
+            addToLayer(bridge.component, bridgeLayer)
         }
 
+        private fun addToLayer(component: Component, layer: Int) {
+            if (renderApi == GraphicsApi.METAL) {
+                // Applying layer on macOS makes our bridge non-transparent
+                // But it draws always on top, so we can just add it as-is
+                // TODO: Figure out why it makes difference in transparency
+                super.add(component, 0)
+            } else {
+                super.setLayer(component, layer)
+                super.add(component)
+            }
+        }
+
+        private val bridgeLayer: Int = 10
+
+        // Place it to top if alpha blending doesn't work
+        private val componentLayer: Int
+            get() = if (isAlphaBlendingWorking) 0 else 20
+
+        // TODO: Support for all platforms
+        private val isAlphaBlendingWorking
+            get() = renderApi == GraphicsApi.METAL // || renderApi == GraphicsApi.DIRECT3D
```

Comment:
Can we refactor so as not to repeat the same code?

Reply:
I'll do it separately, I need to refactor these classes anyway in order to support creating separate scenes for layers. As for this PR - I'm consciously trying not to minimize the number of changes.